applications/StructuralMechanicsApplication/custom_constitutive/user_provided_linear_elastic_law.cpp

```diff
@@ -0,0 +1,285 @@
+    ConstitutiveLaw::Parameters& rParameterValues,
+    const Variable<Matrix>& rThisVariable,
+    Matrix& rValue
+    )
+{
+    if (rThisVariable == CONSTITUTIVE_MATRIX || rThisVariable == CONSTITUTIVE_MATRIX_PK2 || rThisVariable == CONSTITUTIVE_MATRIX_KIRCHHOFF) {
+        this->CalculateElasticMatrix(rValue, rParameterValues);
+    }
+
+    return( rValue );
+}
+
+//*************************CONSTITUTIVE LAW GENERAL FEATURES *************************
+/***********************************************************************************/
+
+template<unsigned int TDim>
+void UserProvidedLinearElasticLaw<TDim>::GetLawFeatures(Features& rFeatures)
+{
+    //Set the type of law
+    rFeatures.mOptions.Set(INFINITESIMAL_STRAINS);
+
+    //Set strain measure required by the consitutive law
+    rFeatures.mStrainMeasures.push_back(StrainMeasure_Infinitesimal);
+    rFeatures.mStrainMeasures.push_back(StrainMeasure_Deformation_Gradient);
+
+    //Set the strain size
+    rFeatures.mStrainSize = GetStrainSize();
+
+    //Set the spacedimension
+    rFeatures.mSpaceDimension = WorkingSpaceDimension();
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<unsigned int TDim>
+int UserProvidedLinearElasticLaw<TDim>::Check(
+    const Properties& rMaterialProperties,
+    const GeometryType& rElementGeometry,
+    const ProcessInfo& rCurrentProcessInfo
+    )
+{
+    KRATOS_CHECK_VARIABLE_KEY(ELASTICITY_TENSOR);
+    return 0;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<unsigned int TDim>
+void UserProvidedLinearElasticLaw<TDim>::CalculateElasticMatrix(
+    Matrix& rConstitutiveMatrix,
+    ConstitutiveLaw::Parameters& rValues
+    )
+{
+    const Properties& r_material_properties = rValues.GetMaterialProperties();
+    rConstitutiveMatrix = r_material_properties[ELASTICITY_TENSOR];
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<unsigned int TDim>
+void UserProvidedLinearElasticLaw<TDim>::CalculatePK2Stress(
+    const Vector& rStrainVector,
+    Vector& rStressVector,
+    ConstitutiveLaw::Parameters& rValues
+    )
+{
+    const Properties& r_material_properties = rValues.GetMaterialProperties();
+    const Matrix C = r_material_properties[ELASTICITY_TENSOR];
+    rStressVector = prod(C, rStrainVector);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<unsigned int TDim>
+void UserProvidedLinearElasticLaw<TDim>::CalculateCauchyGreenStrain(
```

Comment:
This is duplicated in the CL utilities

Reply:
https://github.com/KratosMultiphysics/Kratos/blob/7d87c30d57054bed668e0a6ebb78346792d5a350/applications/StructuralMechanicsApplication/custom_utilities/constitutive_law_utilities.cpp#L373

Reply:
My blocking reason is this, only this

Reply:
> This is duplicated in the CL utilities

Cool. I didn't know about this.